crates/plugins/mediator-coordination/src/didcomm/bridge.rs

```diff
@@ -272,4 +298,114 @@ mod tests {
+                            "crv": "X25519",
+                            "x": "bl_3kgKpz9jgsg350CNuHa_kQL3B60Gi-98WmdQW2h8"
+                        }
+                    }
+                ],
+                "authentication": ["#z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK"],
+                "keyAgreement": ["#z6LSj72tK8brWgZja8NLRwPigth2T9QRiG1uH9oKZuKjdh9p"],
+                "service": []
+            }"##;
+
+        
+    // Resolving Peer DID
+    let did = "did:peer:0z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK";
+    let resolved = resolver.resolve(did).await.unwrap().unwrap();
+
+    // Expected DID Document
+    let expected: serde_json::Value = serde_json::from_str(peer_did_doc).unwrap();
+
+    // Check if resolved DID document matches the expected document
+    assert_eq!(
+        json_canon::to_string(&resolved).unwrap(),
+        json_canon::to_string(&expected).unwrap(),
+        "Resolved DID Document does not match the expected document."
+    );
+}
+
+    #[tokio::test]
+async fn test_local_did_resolver_fails_on_malformed_peer_did() {
+    let diddoc = setup();
+    let resolver = LocalDIDResolver::new(&diddoc);
+
+    // Test cases for malformed Peer DIDs
+    let malformed_dids = [
+        "did:peer:invalid-did-format", // Invalid DID format
+        "did:peer:1234567890123456789012345678901234567890123456789012345678901234567890", // Too long DID
+        "did:peer:0z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK#", // Invalid DID suffix
+    ];
+
+    for did in malformed_dids {
+        let resolved = resolver.resolve(did).await;
+        assert!(matches!(
+            resolved.unwrap_err().kind(),
+            ErrorKind::DIDNotResolved
+        ));
+    }
+}
+
+#[tokio::test]
+async fn test_local_did_resolver_fails_on_unsupported_peer_did_format() {
+    let diddoc = setup();
+    let resolver = LocalDIDResolver::new(&diddoc);
+
+    // Example of an unsupported Peer DID format (incorrect DID syntax or unsupported format)
+    let unsupported_peer_did = "did:peer:unsupported_format";
+
+    // Attempt to resolve the unsupported Peer DID
+    let resolved = resolver.resolve(unsupported_peer_did).await;
+
+    // Assert that the resolver returns a DIDNotResolved error
+    assert!(matches!(
+        resolved.unwrap_err().kind(),
+        ErrorKind::DIDNotResolved
+    ));
+}
+
+#[tokio::test]
+async fn test_local_did_resolver_fails_on_non_existent_peer_did() {
+    let diddoc = setup();
+    let resolver = LocalDIDResolver::new(&diddoc);
+
+    // Example of a Peer DID that doesn't exist in the resolver
+    let non_existent_peer_did = "did:peer:0zNonExistentDID";
+
+    // Attempt to resolve the non-existent Peer DID
+    let resolved = resolver.resolve(non_existent_peer_did).await;
+
+    // Assert that the resolver returns a DIDNotResolved error for the non-existent DID
+    assert!(matches!(
+        resolved.unwrap_err().kind(),
+        ErrorKind::DIDNotResolved
+    ));
+}
+
```

Comment:
Where is the test for successful `did:peer` resolution?

Reply:
I think these is what you are looking for, beginning from line 302 to 352
![Screenshot from 2024-09-26 10-09-50](https://github.com/user-attachments/assets/9b9743a6-a358-4b23-8249-2f55597c86e0)

The test itself was written with the help of @IngridPuppet